From: Simon McVittie <smcv@collabora.com>
Date: Fri, 15 Sep 2017 16:05:48 +0100
Subject: test-help.sh: Skip trivial-httpd if enabled

Because it runs a binary in ${libexecdir}, it will only work if
libostree was already installed, which makes the build-time test fail.
It also doesn't produce the output we expect: its usage mechanism
mentions "ostree-trivial-httpd", not "ostree trivial-httpd".

Signed-off-by: Simon McVittie <smcv@collabora.com>
---
 tests/test-help.sh | 11 ++++++++++-
 1 file changed, 10 insertions(+), 1 deletion(-)

diff --git a/tests/test-help.sh b/tests/test-help.sh
index 75fe0c1..ad74aaf 100755
--- a/tests/test-help.sh
+++ b/tests/test-help.sh
@@ -75,7 +75,16 @@ test_recursive() {
         assert_file_empty out
 
         for subcmd in $builtins; do
-            test_recursive "$cmd $subcmd"
+            case "$subcmd" in
+                (trivial-httpd)
+                    # Skip trivial-httpd if enabled, it doesn't work
+                    # uninstalled (and also doesn't produce the output
+                    # we expect).
+                    ;;
+                (*)
+                    test_recursive "$cmd $subcmd"
+                    ;;
+            esac
         done
     fi
 }
